app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsViewModel.kt

```diff
@@ -0,0 +1,111 @@
+package com.github.se.travelpouch.model.activity.map
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.PolyUtil
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import okhttp3.OkHttpClient
+
+// Define an alias type for a path, which is a list of LatLng points
+typealias Path = List<LatLng>
+
+// Data class to hold the different route between origin and destination
+data class RouteOptions(
+    val origin: LatLng,
+    val destination: LatLng,
+    var routes: List<Path> // List of routes, where each route is a list of LatLng points
+)
+
+/** ViewModel for fetching and managing directions data from the Google Maps Directions API. */
+class DirectionsViewModel(private val repository: DirectionsRepositoryInterface) : ViewModel() {
+
+  // StateFlow to hold all fetched routes for different origin-destination pairs
+  private val _routeOptionsList = MutableStateFlow<List<RouteOptions>>(emptyList())
+  val pathPoints: MutableStateFlow<List<RouteOptions>>
+    get() = _routeOptionsList
```

Comment:
You can change the name to reflect what it holds more clearly as in the suggestion below

```suggestion
    val routeOptions: StateFlow<List<RouteOptions>> get() = _routeOptionsList
```

Reply:
Yes you are right, good catch!